planner/core/plan_cache_lru.go

```diff
@@ -43,11 +44,15 @@ type LRUPlanCache struct {
 	pickFromBucket func(map[*list.Element]struct{}, []*types.FieldType) (*list.Element, bool)
 	// onEvict will be called if any eviction happened, only for test use now
 	onEvict func(kvcache.Key, kvcache.Value)
+
+	// 0 indicates no quota
+	quota uint64
+	guard float64
```

Comment:
Are there any system variables to control this?

Reply:
> Are there any system variables to control this?

yes, to see  `PreparedPlanCacheMaxMemory` and `PreparedPlanCacheMemoryGuardRatio` in `tidb_vars.go`

Reply:
> > Are there any system variables to control this?
> 
> yes, to see `PreparedPlanCacheMaxMemory` and `PreparedPlanCacheMemoryGuardRatio`

sry，`PreparedPlanCacheMaxMemory` is not a system variable,  it is localized, use server-memory-quota as max-plan-cache-memory